src/libraries/System.Private.Xml/src/System/Xml/Xslt/XsltException.cs

```diff
@@ -13,30 +15,30 @@ namespace System.Xml.Xsl
     public class XsltException : SystemException
     {
         private readonly string _res;
-        private readonly string[] _args;
-        private readonly string _sourceUri;
+        private readonly string[]? _args;
+        private readonly string? _sourceUri;
         private readonly int _lineNumber;
         private readonly int _linePosition;
 
         // message != null for V1 & V2 exceptions deserialized in Whidbey
         // message == null for created V2 exceptions; the exception message is stored in Exception._message
-        private readonly string _message;
+        private readonly string? _message;
 
         protected XsltException(SerializationInfo info, StreamingContext context) : base(info, context)
         {
-            _res = (string)info.GetValue("res", typeof(string));
-            _args = (string[])info.GetValue("args", typeof(string[]));
-            _sourceUri = (string)info.GetValue("sourceUri", typeof(string));
-            _lineNumber = (int)info.GetValue("lineNumber", typeof(int));
-            _linePosition = (int)info.GetValue("linePosition", typeof(int));
+            _res = (string)info.GetValue("res", typeof(string))!;
```

Comment:
What ensures that this is non-nullable?

Reply:
this is used for serialization, you will get whatever you wrote.. we should assume payload wasn't tampered.